vms/proposervm/block.go

```diff
@@ -366,15 +366,18 @@ func (p *postForkCommonComponents) verifyPostDurangoBlockDelay(
 	var (
 		blkTimestamp = blk.Timestamp()
 		blkHeight    = blk.Height()
-		currentSlot  = proposer.TimeToSlot(parentTimestamp, blkTimestamp)
 		proposerID   = blk.Proposer()
 	)
+	// populate the slot for the block.
+	blk.slot = new(uint64)
+	*blk.slot = proposer.TimeToSlot(parentTimestamp, blkTimestamp)
```

Comment:
I feel like this would be cleaner:
```golang
var (
		blkTimestamp = blk.Timestamp()
		blkHeight    = blk.Height()
		currentSlot  = proposer.TimeToSlot(parentTimestamp, blkTimestamp)
		proposerID   = blk.Proposer()
	)
	// populate the slot for the block.
	blk.slot = &currentSlot
```
And then we could just use `currentSlot` below like before.